### ISSUE TITLE
Incorrect result for sdpa op

### ISSUE THREAD

**Issue by mmanzoorTT** [tenstorrent/tt-metal#33304]
### Describe the bug

`ttnn.transformer.scaled_dot_product_attention` generates identical results with and without using `scale` attribute. It look likes `scale` attribute is not applied. I have tried multiple `scale` values (0.125, 0.25, 0.5 and few others) but each results in identical output. 

### Steps to reproduce the issue

python script to reproduce the error
```
def test_sdpa():
  device_id = 0
  device = ttnn.open_device(device_id=device_id)
  ttnn.SetDefaultDevice(device)
  try:
    q = torch.randn((1, 16, 32, 64))
    v = torch.randn((1, 16, 32, 64))
    k = torch.randn((1, 16, 32, 64))

    query = ttnn.from_torch(q, ttnn.bfloat16, layout=ttnn.Layout.TILE, device=device)
    key = ttnn.from_torch(k, ttnn.bfloat16, layout=ttnn.Layout.TILE, device=device)
    value = ttnn.from_torch(v, ttnn.bfloat16, layout=ttnn.Layout.TILE, device=device)

    output_0 = ttnn.transformer.scaled_dot_product_attention(query, key, value)
    output_1 = ttnn.transformer.scaled_dot_product_attention(query, key, value, scale=0.125)

    out_0 = ttnn.to_torch(output_0)
    out_1 = ttnn.to_torch(output_1)
    print(f"output without scale: {out_0}")
    print(f"output with scale: {out_1}")

  finally:
      ttnn.close_device(device)
```

### Expected behavior

Generate different result when using `scale` attribute. 

### Please complete the following environment information

OS: Ubuntu 22.04.5 LTS
python version: 3.10.12
tt-metal commit: 62032e61a4

**Comment by tt-rkim** [comment]
Hi, I'm unassigning myself
Why did you assign me to this? I don't work on SDPA @mmanzoorTT

**Comment by tt-rkim** [comment]
@cglagovichTT @esmalTT I've triaged this with `LLM_bug` so it shows up on your boards

**Comment by tt-rkim** [comment]
@ekuslysTT looks like this may be a bug request from Forge team

**Comment by ekuslysTT** [comment]
assigning @alingTT --> missing test coverage

**Comment by mmanzoorTT** [comment]
my bad @tt-rkim but you are one of the code owner of sdpa. So I assigned it to you as well.

**Comment by alingTT** [comment]
```
import torch
import ttnn
from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import (
    comp_allclose,
    comp_pcc,
    comp_and_get_pcc,
)
def test_sdpa():
  device_id = 0
  device = ttnn.open_device(device_id=device_id)
  ttnn.SetDefaultDevice(device)
  try:
    q = torch.randn((1, 16, 32, 64))
    v = torch.randn((1, 16, 32, 64))
    k = torch.randn((1, 16, 32, 64))


    first_scale = 0.5 # <- change scale for first op call
    second_scale = 0.125 # <- change scale for second op call
    # Reshape tensors to [batch, num_heads, seq_len, head_dim] as expected by TTNN and torch
    # Here, torch's scaled_dot_product_attention expects [batch, num_heads, seq, head_dim]
    # TTNN should use the same shape

    query = ttnn.from_torch(q, ttnn.bfloat16, layout=ttnn.Layout.TILE, device=device)
    key = ttnn.from_torch(k, ttnn.bfloat16, layout=ttnn.Layout.TILE, device=device)
    value = ttnn.from_torch(v, ttnn.bfloat16, layout=ttnn.Layout.TILE, device=device)

    # TTNN scaled dot product attention with two scales
    output_0 = ttnn.transformer.scaled_dot_product_attention(query, key, value, scale=first_scale)
    output_1 = ttnn.transformer.scaled_dot_product_attention(query, key, value, scale=second_scale)

    out_0 = ttnn.to_torch(output_0)
    out_1 = ttnn.to_torch(output_1)

    # Torch version for 0.55 scale
    torch_out_0 = torch.nn.functional.scaled_dot_product_attention(
        q, k, v, attn_mask=None, dropout_p=0.0, is_causal=True, scale=first_scale
    )
    # Torch version for 0.125 scale
    torch_out_1 = torch.nn.functional.scaled_dot_product_attention(
        q, k, v, attn_mask=None, dropout_p=0.0, is_causal=True, scale=second_scale
    )

    out_pass_0, out_pcc_0 = comp_pcc(out_0, torch_out_0, 0.999)
    out_pass_1, out_pcc_1 = comp_pcc(out_1, torch_out_1, 0.999)

    print(f"PCC output_0 vs torch_out_0 (scale={first_scale}): {out_pcc_0}")
    print(f"PCC output_1 vs torch_out_1 (scale={second_scale}): {out_pcc_1}")

    # Show where output_0 is different from output_1
    difference = out_0 != out_1
    if difference.any():
        differing_indices = difference.nonzero(as_tuple=True)
        print("Indices where output_0 is different from output_1:", differing_indices)
        print("output_0 values at differing indices:", out_0[differing_indices])
        print("output_1 values at differing indices:", out_1[differing_indices])
    else:
        print("output_0 and output_1 are identical at all positions.")

  finally:
      ttnn.close_device(device)

if __name__ == "__main__":
  test_sdpa()
```

Hey so if no scale is passed, scale automatically computed is 1/sqrt(dim) = 1/sqrt(64) =0.125, so with the test case you provided, passing no scale = 0.125 so both op calls have the same scale.

If you run this test by changing the first op call's scale to for example 0.55 vs another op call with 0.125, you can see that the tensors are different. I tried 0.25 vs 0.125, 0.5 vs 0.125 they all have different tensor outputs, and for all combinations the output has 0.999 PCC with the torch output.

0.125 vs 0.25:
```
PCC output_0 vs torch_out_0 (scale=0.25): Max ATOL Delta: 0.125, Max RTOL Delta: 232.0, PCC: 0.9998411144953587
PCC output_1 vs torch_out_1 (scale=0.125): Max ATOL Delta: 0.109375, Max RTOL Delta: 205.0, PCC: 0.9998262138264737
Indices where output_0 is different from output_1: (tensor([0, 0, 0,  ..., 0, 0, 0]), tensor([ 0,  0,  0,  ..., 15, 15, 15]), tensor([ 1,  1,  1,  ..., 31, 31, 31]), tensor([ 0,  1,  2,  ..., 61, 62, 63]))
output_0 values at differing indices: tensor([0.1797, 0.8945, 0.7812,  ..., 0.1162, 0.3945, 0.3398],
       dtype=torch.bfloat16)
output_1 values at differing indices: tensor([0.3770, 0.7227, 0.6016,  ..., 0.0510, 0.2930, 0.1641],
       dtype=torch.bfloat16)
```

0.125 vs 0.5:
```
PCC output_0 vs torch_out_0 (scale=0.5): Max ATOL Delta: 0.1875, Max RTOL Delta: 1104.0, PCC: 0.9998377552259141
PCC output_1 vs torch_out_1 (scale=0.125): Max ATOL Delta: 0.109375, Max RTOL Delta: 592.0, PCC: 0.9998086952762368
Indices where output_0 is different from output_1: (tensor([0, 0, 0,  ..., 0, 0, 0]), tensor([ 0,  0,  0,  ..., 15, 15, 15]), tensor([ 1,  1,  1,  ..., 31, 31, 31]), tensor([ 0,  1,  2,  ..., 61, 62, 63]))
output_0 values at differing indices: tensor([-0.4727,  1.3828,  1.2422,  ...,  0.4121,  0.2988, -0.0608],
       dtype=torch.bfloat16)
output_1 values at differing indices: tensor([-0.6172,  1.4922,  1.1484,  ...,  0.1611,  0.2490, -0.1147],
       dtype=torch.bfloat16)
```

Please let me know if you have trouble reproducing this, because just based on this I dont think this is a legitimate bug unless there is something I missed thanks

**Comment by alingTT** [comment]
Going to close this issue, if there are other issues with SDPA please file a different issue.
